### PR TITLE
ZBS320/profile plaster

### PR DESCRIPTION
**Pull request by zack-erisyon**
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip
- wip

### CODE REVIEW

File: plaster/tools/zlog/zlog.py [hunk 1]
```diff
@@ -290,6 +291,56 @@ def metrics(message, log=None, **kwargs):
...
 
 
+prof_stack = []
+
+
+def prof_start(block_name, log=None, stacklevel=2, **kwargs):
+    global prof_stack
+
+    if log is None:
+        log = logging.getLogger("plaster_profile")
```
Comment (danwinkler) [inline plaster/tools/zlog/zlog.py:301]: Why is this logger not in the plaster namespace (`plaster.zlog.profile`)? Seems like the default behavior we'd want it for it to log like plaster unless we specifically override it (as shown by the added config lines you've added to all of the logging.yaml files).

File: plaster/tools/zlog/zlog.py [hunk 2]
```diff
@@ -290,6 +291,56 @@ def metrics(message, log=None, **kwargs):
...
+    elapsed = now - start
+
+    kwargs_per_sec = {f"{key}_per_sec": val / elapsed for key, val in kwargs.items()}
+
+    kwargs_strs = [
+        f"{key}={val:2.2f}" if isinstance(val, (int, float)) else f"{key}={val}"
+        for key, val in {**kwargs, **kwargs_per_sec}.items()
+    ]
+
+    log.info(
```
Comment (danwinkler) [inline plaster/tools/zlog/zlog.py:324]: Might make sense to have the level be configurable at some point but this is fine for now

**Review by danwinkler — CHANGES_REQUESTED** [review]
Not a fan of having profile logs end up under `plaster_profile`, and having to add new, identical configuration everywhere. Seems better to just have it under plaster, leave all of the logging.yamls how they are, and if we want to tweak the profile logs specifically we can still do that. 

Otherwise lgtm

File: plaster/tools/zlog/zlog.py (hunk 1 above)
Comment (zack-erisyon) [reply plaster/tools/zlog/zlog.py]: done

**Pull request merged**